### PR TITLE
Make as_manager method use proper generic

### LINKED ISSUE #324: Make as_manager() in queryset generic instead of using Any
# Bug report

## What's wrong
`as_manager()` on querysets currently uses `Any` instead of `_T`.

Like so:

```python
# django-stubs/db/models/query.pyi
    @classmethod
    def as_manager(cls) -> Manager[Any]: ...
```
## How is that should be
Looks like a simple switch to _T fixes it:

```python
# django-stubs/db/models/query.pyi
    @classmethod
    def as_manager(cls) -> Manager[_T]: ...
```

See https://docs.djangoproject.com/en/3.0/topics/db/managers/#creating-a-manager-with-queryset-methods for more info.

### PR DESCRIPTION
**Pull request by silviogutierrez**
- Closes #324

### CODE REVIEW

File: django-stubs/db/models/query.pyi [hunk 1]
```diff
@@ -41,7 +41,7 @@ class _BaseQuerySet(Generic[_T], Sized):
         hints: Optional[Dict[str, models.Model]] = ...,
     ) -> None: ...
     @classmethod
-    def as_manager(cls) -> Manager[Any]: ...
+    def as_manager(cls) -> Manager[_T]: ...
```
Comment (sobolevn) [inline django-stubs/db/models/query.pyi:44]: Can you please also test it? We have a lot of bugs in QuerySet / Manager topics.
And we want to be sure it is as stable as possible.

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks a lot for the fix!

**Comment by lithammer** [comment]
Possible duplicate work in https://github.com/typeddjango/django-stubs/pull/290.

**Comment by silviogutierrez** [comment]
@sobolevn : forgive me if I missed it, but is there a guide to running or developing locally? I'm trying to run tests and doing it through CI passes really takes a while. Got a lot of other fixes too that I'd love to contribute back.

I tried replicating the Travis steps but no luck.

An example here: https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/unit-tests/#running-the-unit-tests

**Comment by sobolevn** [comment]
Yes, we have a really short guide: https://github.com/typeddjango/django-stubs/blob/master/CONTRIBUTING.md

It is not perfect and I am open for any feedback!

**Comment by RJPercival** [comment]
Has this been abandoned?

**Comment by sciyoshi** [comment]
We are waiting for this change as well as it is blocking us upgrading to Python 3.9. Is this still only waiting on tests @sobolevn?

**Comment by sciyoshi** [comment]
It does look like the PR includes a test already - I'm happy to help if there's anything needed to move this forward.

**Comment by sobolevn** [comment]
I would love to merge this after the CI is fixed.

**Comment by sciyoshi** [comment]
After investigating a bit more - looks like this simple approach doesn't quite work and it will probably need some of the more advanced machinery from #272 or #290. I'd recommend closing this PR in favor of one of the other two unless @silviogutierrez has some further input on it.

**Comment by silviogutierrez** [comment]
@sciyoshi : yea, #272 and #290 are much better solutions. Thanks all.

**Pull request closed without merging**